homeassistant/components/feedreader/__init__.py

```diff
@@ -62,236 +49,13 @@ async def async_setup(hass: HomeAssistant, config: ConfigType) -> bool:
-        self._scan_interval = scan_interval
-        self._max_entries = max_entries
-        self._feed: feedparser.FeedParserDict | None = None
-        self._firstrun = True
-        self._storage = storage
-        self._last_entry_timestamp: struct_time | None = None
-        self._has_published_parsed = False
-        self._has_updated_parsed = False
-        self._event_type = EVENT_FEEDREADER
-        self._feed_id = url
-
-    @callback
-    def async_setup(self) -> None:
-        """Set up the feed manager."""
-        self._hass.bus.async_listen_once(EVENT_HOMEASSISTANT_START, self._async_update)
-        async_track_time_interval(
-            self._hass, self._async_update, self._scan_interval, cancel_on_shutdown=True
-        )
-
-    def _log_no_entries(self) -> None:
-        """Send no entries log at debug level."""
-        _LOGGER.debug("No new entries to be published in feed %s", self._url)
-
-    async def _async_update(self, _: datetime | Event) -> None:
-        """Update the feed and publish new entries to the event bus."""
-        last_entry_timestamp = await self._hass.async_add_executor_job(self._update)
-        if last_entry_timestamp:
-            self._storage.async_put_timestamp(self._feed_id, last_entry_timestamp)
-
-    def _update(self) -> struct_time | None:
-        """Update the feed and publish new entries to the event bus."""
-        _LOGGER.debug("Fetching new data from feed %s", self._url)
-        self._feed = feedparser.parse(
-            self._url,
-            etag=None if not self._feed else self._feed.get("etag"),
-            modified=None if not self._feed else self._feed.get("modified"),
-        )
-        if not self._feed:
-            _LOGGER.error("Error fetching feed data from %s", self._url)
-            return None
-        # The 'bozo' flag really only indicates that there was an issue
-        # during the initial parsing of the XML, but it doesn't indicate
-        # whether this is an unrecoverable error. In this case the
-        # feedparser lib is trying a less strict parsing approach.
-        # If an error is detected here, log warning message but continue
-        # processing the feed entries if present.
-        if self._feed.bozo != 0:
-            _LOGGER.warning(
-                "Possible issue parsing feed %s: %s",
-                self._url,
-                self._feed.bozo_exception,
-            )
-        # Using etag and modified, if there's no new data available,
-        # the entries list will be empty
-        _LOGGER.debug(
-            "%s entri(es) available in feed %s",
-            len(self._feed.entries),
-            self._url,
-        )
-        if not self._feed.entries:
-            self._log_no_entries()
-            return None
-
-        self._filter_entries()
-        self._publish_new_entries()
-
-        _LOGGER.debug("Fetch from feed %s completed", self._url)
+    await asyncio.gather(*[feed.async_refresh() for feed in feeds])
```

Comment:
Any exception coming out of these are not handled. Consider using the first refresh method if you want to fail setup if any feed is unavailable (although that might not be what the user wants)

Reply:
we've no config entry, yet, so the `async_config_entry_first_refresh` wouldn't work properly in this case